liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,256 @@
+        else:
+            _LOGGER.debug('pump_mode %s', pump_mode)
+            self._write([_PRO_CMD_WRITE_PUMP_MODE, _PRO_PUMP_MODES.index(pump_mode)])
+            self._end_transaction_and_read(5)
+
+    def get_status(self, **kwargs):
+        """Get a status report.
+        Returns a list of `(property, value, unit)` tuples.
+        """
+        status = []
+        ### Can some Pro devices have more than one temp sensor?   OCL seems to code for that?
+        self._write([_PRO_CMD_READ_AIO_TEMP])
+        msg = self._end_transaction_and_read(6)
+        aio_temp = msg[3] + msg[4]/10
+        status.append(('Liquid temperature', aio_temp, '°C'))
+        speeds = self._get_fan_speeds()
+        for reading in speeds:
+            status.append(('Fan {} speed'.format(reading[0]), reading[1] | 0, 'rpm'))
+        self._write([_PRO_CMD_READ_PUMP_MODE])
+        msg = self._end_transaction_and_read(4)
+        pump_mode = msg[3]
+        status.append(('Pump mode', _PRO_PUMP_MODES[pump_mode], ""))
+        self._write([_PRO_CMD_READ_PUMP_SPEED])
+        msg = self._end_transaction_and_read(5)
+        pump_speed = (msg[3] << 8) + msg[4]
+        status.append(('Pump speed', pump_speed | 0, 'rpm'))
+        self._write([_PRO_CMD_READ_FIRMWARE])
+        msg = self._end_transaction_and_read(7)
+        firmware = '{}.{}.{}.{}'.format(*tuple(msg[3:7]))
+        status.append(('Firmware version', firmware, ''))
+        return status
+
+    def set_color(self, channel, mode, colors, time_per_color=1, time_off=None,
+                  alert_threshold=_HIGH_TEMPERATURE, alert_color=[255, 0, 0],
+                  speed=3, temps=['35', '45', '55'], **kwargs):
+        """Set the color mode for a specific channel."""
+        if mode not in _PRO_COLOR_CHANGE_MODES:
+            raise KeyError('Unknown lighting mode {}'.format(mode))
+        colors = list(colors)
+        _LOGGER.debug('color count = %d', len(colors))
+        _LOGGER.debug('color change speed %d', int(speed))
+        if mode == 'alert':
+            self._write([0x5f, int(temps[0]), 0x00, int(temps[1]), 0x00, int(temps[2]), 0x00] + colors[0] + colors[1] + colors[2])
+            self._end_transaction_and_read(6)
+            self._write([0x5E, 0x01])
+            self._end_transaction_and_read(3)
+            return
+        if mode == 'fixed':
+            colors = [colors[0], colors[0]]
+        if mode != 'rainbow':
+            self._write_color(colors)
+        if mode != 'fixed':
+            self._write_color_change_speed(mode, speed)
+        self._write(_PRO_COLOR_CHANGE_MODES[mode])
+        self._end_transaction_and_read(3)
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        """Set channel to follow a speed duty profile."""
+        if channel == 'pump':
+            raise ValueError('set_speed_profile not implemented for the pump on CorsairAsetekPro')
+        adjusted = self._prepare_profile(profile, 0, 100, _MAX_PROFILE_POINTS)
```

Comment:
The channel name should probably be validated before computing the profile.  Any comparisons should be case insensitive, but if the channel is neither `fan` nor a valid `fan[0-9]` value, you should raise a `ValueError`.

Note: not all liquidctl errors are specified yet, so for now and if convenient you can also let a `KeyError` or `IndexError` bubble up.  Nevertheless a customized `ValueError` message would be preferable.

Reply:
I've used a `KeyError` with a custom message for now

Reply:
That's better, but:
- `foo` raises `KeyError`
- `fanfoo` raises `ValueError` (from `int(...)`)

More importantly, and I missed this before, `fan4` (or `fan99`) will *not* raise, and result in unsound writes.

Reply:
Regarding your question about making `_CommonAsetekDriver` public, I see your point: inheriting from a private class in another module is, err, wrong.

But I think the proper solution is to clean up, document, test and move `prepare_profile` to `liquidctl.util`, and stop inheriting from any `asetek` classes (instead inheriting from `liquidctl.usb.UsbDriver` directly).

Reply:
I've changed this to raise `ValueError`s. I've also added a check to make sure the fan specified exists (so `fan99` should now error).